src/tray.c

```diff
@@ -0,0 +1,162 @@
+ *
+ * You must obey the GNU General Public License in all respects for all of the
+ * code used other than OpenSSL. If you modify file(s) with this exception, you
+ * may extend this exception to your version of the file(s), but you are not
+ * obligated to do so. If you do not wish to do so, delete this exception
+ * statement from your version. If you delete this exception statement from all
+ * source files in the program, then also delete it here.
+ *
+ */
+
+#include "prof_config.h"
+
+#ifdef PROF_HAVE_GTK
+#include <gtk/gtk.h>
+#include <glib.h>
+#include <glib/gstdio.h>
+
+#include "tray.h"
+#include "window_list.h"
+
+static GtkStatusIcon *prof_tray = NULL;
+static GString *icon_filename = NULL;
+static GString *icon_msg_filename = NULL;
+static gint unread_messages;
+static gboolean shutting_down;
+static guint timer;
+
+/* {{{ Privates */
+
+/*
+ * Get icons from installation share folder or (if defined) .locale user's folder
+ *
+ * As implementation, looking through all the entries in the .locale folder is chosen.
+ * While useless as now, it might be useful in case an association name-icon is created.
+ * As now, with 2 icons only, this is pretty useless, but it is not harming ;)
+ *
+ */
+static void _get_icons(void)
+{
+    GString *icons_dir =  NULL;
+
+#ifdef ICONS_PATH
+
+    icons_dir = g_string_new(ICONS_PATH);
+    icon_filename = g_string_new(icons_dir->str);
+    icon_msg_filename = g_string_new(icons_dir->str);
+    g_string_append(icon_filename, "/proIcon.png");
+    g_string_append(icon_msg_filename, "/proIconMsg.png");
+    g_string_free(icons_dir, true);
+
+#endif /* ICONS_PATH */
+
+    gchar *xdg_config = xdg_get_config_home();
+    icons_dir = g_string_new(xdg_config);
+    g_free(xdg_config);
+    g_string_append(icons_dir, "/profanity/icons");
+    GError *err = NULL;
+    if (!g_file_test(icons_dir->str, G_FILE_TEST_IS_DIR)) {
+        return;
+    }
+    GDir *dir = g_dir_open(icons_dir->str, 0, &err);
+    if (dir) {
+        GString *name = g_string_new(g_dir_read_name(dir));
+        while (name->len) {
+            if (g_strcmp0("proIcon.png", name->str) == 0) {
+                if (icon_filename) {
+                    g_string_free(icon_filename, true);
+                }
+                icon_filename = g_string_new(icons_dir->str);
+                g_string_append(icon_filename, "/proIcon.png");
+            } else
+            if (g_strcmp0("proIconMsg.png", name->str) == 0){
+                if (icon_msg_filename) {
+                    g_string_free(icon_msg_filename, true);
+                }
+                icon_msg_filename = g_string_new(icons_dir->str);
+                g_string_append(icon_msg_filename, "/proIconMsg.png");
+            }
+            g_string_free(name, true);
+            name = g_string_new(g_dir_read_name(dir));
+        }
+        g_string_free(name, true);
+    } else {
+        fprintf (stderr, "Unable to open dir: %s\n", err->message);
```

Comment:
`stderr` is piped to the profanity logs, but if you include `log.h` you can use `log_error()` which uses the profanity log handling code instead. It takes a string format and varargs and doesn't need a newline.